### ISSUE TITLE
Multiple items are not getting uploaded for Dropbox and Internet Archive

### ISSUE THREAD

**Issue by purvi-ranawat** [OpenArchive/Save-app-ios#196]
When we select multiple images and upload all of them at once, only the first item is getting uploaded and for rest of the images the process keeps going on.

Steps to reproduce:

Select 5 or more images from photo gallery
Upload to Dropbox or Internet Archive servers

Expected result: All items gets uploaded one by one

Actual result: Only first item gets upload and for rest of the images the process keeps going on and then they are not uploaded

Tested on iPhone XR, 15.4.1, app version 2.7.1 (58)

Note: Issue exists only on Internet Archive and Dropbox, it worked fine on Nextcloud server.

Below is the screenshot : 

![WhatsApp Image 2022-05-09 at 9 43 13 PM](https://user-images.githubusercontent.com/8190141/167454242-cf8d5c01-bda1-4ecf-a154-bb5371b6145b.jpeg)

**Comment by johnhess** [comment]
I'm having the same issue (tested with dropbox), even with the app in the foreground. 

When I kill and restart the app, the top most item "finishes" and the next one starts.  Then it too stalls as the progress hits 100%.  At this point, the file and metadata have been safely delivered to Dropbox.

**Comment by johnhess** [comment]
Similarly, hitting "Stop" after the circular progress animation reaches 100% causes the app to move on to the next upload.

**Comment by tladesignz** [comment]
Seems, I broke the Dropbox upload slightly, when doing all this background rework. Fixed now.

For Internet Archive - cannot reproduce. This works just fine for me, as expected, since it actually uses the same code as the Nextcloud support.

Please test and throw back, if this still happens to you!

Fix will be contained in TestFlight build 60!

**Comment by purvi-ranawat** [comment]
@tladesignz @johnhess For Version 61, multiple video files are not getting uploaded on Dropbox. If I select multiple images with smaller file size, it uploads successfully.
![WhatsApp Image 2022-05-29 at 11 16 31 AM](https://user-images.githubusercontent.com/8190141/170854244-7e4919ee-4649-4c5d-b533-33e610f53bce.jpeg)

**Comment by tladesignz** [comment]
> @tladesignz @johnhess For Version 61, multiple video files are not getting uploaded on Dropbox. If I select multiple images with smaller file size, it uploads successfully.

Yes, this is a limitation of the current Dropbox support. If you hit the red (!), the error message given will explain that. The limit is 150 MByte.

There is a complicated support for larger files which includes using some "upload session", but I didn't use that at first, as it would clearly not work in the background and it was a whole lot to implement.
We probably should revisit this, now that we positively know that we cannot ever support background upload anyway.

I suggest to create a new issue for this, if you want this, @johnhess!
